libraries/aws_backend.rb

```diff
@@ -148,6 +148,18 @@ def initialize(opts)
       client_args[:client_args][:region] = opts[:aws_region] if opts[:aws_region]
       # below allows each resource to optionally and conveniently set an endpoint
       client_args[:client_args][:endpoint] = opts[:aws_endpoint] if opts[:aws_endpoint]
+      # below allows each resource to optionally and conveniently set max_retries
+      if opts[:retry_limit]
+        client_args[:client_args][:retry_limit] = opts[:retry_limit].to_i
+      elsif ENV['AWS_RETRY_LIMIT']
+        client_args[:client_args][:retry_limit] = ENV['AWS_RETRY_LIMIT'].to_i
```

Comment:
on windows someone might want to set `aws_retry_limit` - would be good to check for a lowercase key in ENV to support this